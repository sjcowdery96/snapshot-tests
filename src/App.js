import logo from './logo.svg';
import './App.css';
import { useEffect, useState } from 'react';
import MyGitHubCard from './components/GitHubCard';

function App() {
  //gets my github info
  const [myGitName, getGitName] = useState('')
  const [myGitUrl, getGitURL] = useState('')
  const [myGitFace, getGitFace] = useState('')
  const [myGitBio, getGitBio] = useState('')
  //creates our useEffect to fetch once and never againg -- empty dependency array
  useEffect(() => {
    fetch('https://api.github.com/users/sjcowdery96')
      .then(res => res.json())
      .then(data => {
        getGitName(data.name)
        getGitURL(data.html_url)
        getGitFace(data.avatar_url)
        getGitBio(data.bio)
      })

  }, [])

  return (
    <div className="App">
      <MyGitHubCard myBio={myGitBio} myName={myGitName} myface={myGitFace} myURL={myGitUrl}></MyGitHubCard>
    </div>
  );
}

export default App;
