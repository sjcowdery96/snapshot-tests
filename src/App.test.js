import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer'

import App from './App';

test('renders my Github Bio', async () => {
  //can't get this test to work even when my actual bio renders...
  //fetch.mockResponseOnce(JSON.stringify({ myBio: "Let's get technical." }))
  render(<App />);
  const myBio = await waitFor(() => screen.getByRole('button', screen.getByRole('heading', { level: 2 })))
  expect(myBio).toHaveTextContent("Let's get technical.");
});

test('renders a snapshot', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})


