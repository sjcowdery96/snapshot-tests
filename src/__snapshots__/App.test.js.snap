// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <div
    className="card"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <img
      className="card-img-top"
      src=""
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      />
      <p
        className="card-text"
      />
      <p
        className="card-text"
      >
        <i>
          data pulled from 
        </i>
      </p>
      <a
        className="btn btn-primary"
        disabled={false}
        href=""
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        visit profile
      </a>
    </div>
  </div>
</div>
`;
